src/ast.rs

```diff
@@ -549,11 +555,84 @@ impl Module {
 // Virtual modules cannot be compile targets
 // Please see doc/modules.md for more info
 impl Virtualable for Module {
-    fn set_virtual(&mut self) {
+    fn set_virtual(&mut self, _range: Range<usize>) -> Result<(), ParseErrorMsg> {
         self.is_virtual = true;
+        Ok(())
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct System {
+    pub name: CascadeString,
+    pub is_virtual: bool,
+    pub annotations: Annotations,
+    pub modules: Vec<CascadeString>,
+    pub configurations: Vec<LetBinding>,
+}
+
+impl System {
+    pub fn new(name: CascadeString) -> Self {
+        System {
+            name,
+            is_virtual: false,
+            annotations: Annotations::new(),
+            modules: Vec::new(),
+            configurations: Vec::new(),
+        }
+    }
+
+    pub fn set_fields(mut self, input: Vec<SystemBody>) -> Self {
+        for i in input {
+            match i {
+                SystemBody::Mod(m) => {
+                    self.modules.push(m);
+                }
+                SystemBody::Config(l) => {
+                    self.configurations.push(l);
+                }
+            }
+        }
+        // Insert the default configurations if they were not provided
+        if !self
```

Comment:
I'd rather have this just loop over a list of [(config_name, default_value), ...].  Similar to how the get_built_in_types_map() function works.  You can even define the default in constants.rs like with get_built_in_types_map().